lib/exercism/user_progression.rb

```diff
@@ -2,15 +2,36 @@
 class UserProgression
 
   def self.user_progress(user)
-    track_complete_exercises = Hash.new 0
-    user.exercises.each{|exe| track_complete_exercises[exe.problem.language] += 1}
+    track_complete_exercises = {}
+    user.exercises.each do |exe|
+      track_complete_exercises[exe.problem.language] ||= LanguageProgress.new exe.problem.language
+      track_complete_exercises[exe.problem.language].add_user_exercise exe
+    end
     tracks = track_count
-    track_complete_exercises.each{|k, v| track_complete_exercises[k] = [v, tracks[k]]}
-    track_complete_exercises
+    track_complete_exercises.collect do |lang, language_progress| 
+      language_progress.tap{ |lang_prog| lang_prog.language_track = tracks[lang] }
+    end
   end
 
+  class LanguageProgress
+    attr_accessor :language, :user_exercises, :last_updated, :language_track
```

Comment:
@paradime 
have you thought about making some of these only readable? i.e. :user_exercises or :language_track?